src/EventStore.Core/TransactionLog/LogRecords/PrepareLogRecord.cs

```diff
@@ -158,7 +159,8 @@ public class PrepareLogRecord : LogRecord, IEquatable<PrepareLogRecord>, IPrepar
 			TimeStamp = timeStamp;
 			EventType = eventType ?? string.Empty;
 			_eventTypeSize = eventTypeSize;
-			Data = Flags.HasFlag(PrepareFlags.IsRedacted) ? NoData : data;
+			_data = data;
+			Data = Flags.HasFlag(PrepareFlags.IsRedacted) ? NoData : _data;
 			Metadata = metadata;
 			if (InMemorySize > TFConsts.MaxLogRecordSize) throw new Exception("Record too large.");
```

Comment:
the following internal methods also need to be adjusted to use `_dataOnDisk` instead of `Data`
- InMemorySize - this one may seem counterintuitive but its actually used to check if the `PrepareLogRecord` doesn't exceed the max log record size, so we do need `_dataOnDisk` here
- SizeOnDisk
- Equals
- GetHashCode

Reply:
I guess we can use `_dataOnDisk` instead of `Data` for the methods `InMemorySize`, `SizeOnDisk`, and `GetHashCode`. For the `Equals` method, it is returning true or false based on whether 2 `PrepareLogRecord` objects are same or not (whether they have same object parameters or not). So maybe in the `Equals` method, instead of replacing `Data` with `_dataOnDisk`, we can keep `Data` and also add the equality check for `_dataOnDisk`.

Reply:
sounds fair enough, `Equals` is used only in tests anyway so it shouldn't have any performance implications